netbox/netbox/navigation_menu.py

```diff
@@ -408,18 +408,28 @@ def get_model_buttons(app_label, model_name, actions=('add', 'import')):
 if registry['plugins']['menu_items']:
     plugin_menu_groups = []
 
-    for plugin_name, items in registry['plugins']['menu_items'].items():
-        plugin_menu_groups.append(
-            MenuGroup(
-                label=plugin_name,
-                items=items
+    for plugin_name, data in registry['plugins']['menu_items'].items():
+        if data['header']:
+            menu_groups = [MenuGroup(label=plugin_name, items=data["items"])]
```

Comment:
We should probably give some more thought to handling groups within a top-level menu. Ideally, plugins should be able to declare their own groups of items. I'm going to think about this some more.